website/client/js/controllers/userCtrl.js

```diff
@@ -15,6 +15,13 @@ habitrpg.controller("UserCtrl", ['$rootScope', '$scope', '$location', 'User', '$
     $scope.$watch('_editing.profile', function(value){
       if(value === true) $scope.editingProfile = angular.copy(User.user.profile);
     });
+    
+    $scope.isBasicAnimal = function(name, type) {
+      var tmp = name.split('-');
+      $scope[type + "Egg"] = tmp[0];
+      $scope[type + "Potion"] = tmp[1];
+      return Content.hatchingPotions[$scope[type + "Potion"]] ? true : false;
+    }
```

Comment:
Code that does the same thing for the same reason in two different places shouldn't be written twice.  Write it somewhere else and use it twice instead.